class-two-factor-core.php

```diff
@@ -1873,6 +1873,17 @@ public static function user_two_factor_options_update( $user_id ) {
 					) );
 				}
 			}
+
+			// Destroy other sessions if we've activated a new provider.
+			if ( array_diff( $enabled_providers, $existing_providers ) ) {
```

Comment:
```suggestion
			// Destroy other sessions if setup 2FA for the first time, or deactivated a provider
			if (
				// No providers, enabling one (or more)
				( ! $existing_providers && $enabled_providers ) ||
				// Has providers, and is disabling one (or more)
				( $existing_providers && array_diff( $existing_providers, $enabled_providers ) )
			) {
```

Based on @iandunn's comment above, this would match as requested, I think.